### DISCUSSION TITLE
Gerar hash a partir do c#

### DISCUSSION THREAD

**Discussion by FernandoManuel6107** [assoft-portugal/SAF-T-AO#164] [Q&A] [answered]
Olá ilustres,

Gostaria que alguém me ajudasse com um exemplo ou um link com passos para gerar o hash dentro de uma aplicação c#.

Já segui vários exemplos, dentre os quais a biblioteca Chilkat, que funcionou muito bem mas só que é paga.

Agradeço

@cryptolopes

**Comment by cryptolopes** [comment, marked as answer]
- [RSACryptoServiceProvider.SignHash Method](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsacryptoserviceprovider.signhash?view=net-5.0)
- [Cryptographic Signatures](https://docs.microsoft.com/en-us/dotnet/standard/security/cryptographic-signatures)
- [Create RSA-SHA1 signature](https://stackoverflow.com/a/49082869)

**Reply by FernandoManuel6107** [reply]
Muito obrigado
